codes/data/italy/italy-download-cases-merge-policies.py

```diff
@@ -38,534 +49,294 @@
+		        ), col] = np.nan
+		        
+		    return adm_cases
+
+		# Fill in nulls for missing adm1 data, in the adm1 dataset
+		for date, adm1 in adm1_days_missing:
+		    adm1_cases = fill_missing_as_null(adm1_cases, date, adm1, 'adm1_name')
+		        
+		# Fill in nulls for missing adm1 data, in the adm2 dataset
+		for date, adm1 in adm1_days_missing:
+		    adm2_cases = fill_missing_as_null(adm2_cases, date, adm1, 'adm1_name')
+
+		# Fill in nulls for missing adm2 data, in the adm2 dataset
+		for date, adm2 in adm2_days_missing:
+		    adm2_cases = fill_missing_as_null(adm2_cases, date, adm2, 'adm2_name')
+
+		return adm1_cases, adm2_cases
+
+	adm2_cases = impute_day1_adm2_cases(adm1_cases, adm2_cases)
+	adm1_cases, adm2_cases = code_as_null_where_missing(adm1_cases, adm2_cases)
+
+	def impute_each_cumulative_column(adm_cases, adm_id_col):
+		# Impute any cumulative totals in adm1
+		adm_cases_cum_cols = [col for col in adm_cases.columns if col.startswith(cumulative_prefix)]
+		for src_col in adm_cases_cum_cols:
+		    dst_col = src_col + imputed_suffix
+		    adm_cases = cimpute.impute_cumulative_df(adm_cases, src_col, dst_col, adm_id_col)
+
+		return adm_cases
+
+	# #### Impute all cumulative totals in imputed column, fill as null where cumulative totals fall in source column
+	adm1_cases = impute_each_cumulative_column(adm1_cases, 'adm1_name')
+	adm2_cases = impute_each_cumulative_column(adm2_cases, 'adm2_id')
+
+	# #### Save processed health data to `interim` folder
+	path_italy_interim_province.parent.mkdir(parents=True, exist_ok=True)
+	adm1_cases.to_csv(path_italy_interim_region, index=False)
+	adm2_cases.to_csv(path_italy_interim_province, index=False)
+
+	return adm1_cases, adm2_cases
+
+def read_policies():
+	policies = pd.read_csv(path_italy_policies)
+
+	# Map some regions/provinces in policy dataset to corresponding names in health data
+	replace_dict = {
+	    'Lombardy':'Lombardia',
+	    'Piedmont':'Piemonte',
+	    'Emilia-Romagna':'Emilia Romagna',
+	    'Padua':'Padova',
+	    'Venice':'Venezia',
+	    'Pesaro and Urbino':'Pesaro e Urbino',
+	    'Apulia':'Puglia', 
+	    "Vo'Eugane":'Vò',
+	}
+
+	# Standardize naming between policy and health data
+	policies['adm1_name'] = policies['adm1_name'].replace(replace_dict)
+	policies['adm2_name'] = policies['adm2_name'].replace(replace_dict)
+	policies['adm3_name'] = policies['adm3_name'].replace(replace_dict)
+
+	# Clean data
+	policies['date_start'] = pd.to_datetime(policies['date_start'])
+	policies['date_end'] = pd.to_datetime(policies['date_end'])
+	policies['policy'] = policies['policy'].str.strip()
+
+	# Set default values for null fields
+	policies['adm0_name'] = policies['adm0_name'].fillna('Italy')
+	policies['adm1_name'] = policies['adm1_name'].fillna('All')
+	policies['adm2_name'] = policies['adm2_name'].fillna('All')
+	policies['adm3_name'] = policies['adm3_name'].fillna('All')
+
+	# Remove any duplicates, grouping on relevant columns
+	policies = policies[
```

Comment:
should this raise a warning so that folks know to go remove the duplicates from the gsheet?